library/alloc/src/collections/btree/node.rs

```diff
@@ -120,11 +121,11 @@ struct BoxedNode<K, V> {
 
 impl<K, V> BoxedNode<K, V> {
     fn from_leaf(node: Box<LeafNode<K, V>>) -> Self {
-        BoxedNode { ptr: Box::into_unique(node).0 }
+        BoxedNode { ptr: Unique::from(Box::leak(node)) }
     }
 
     fn from_internal(node: Box<InternalNode<K, V>>) -> Self {
-        BoxedNode { ptr: Unique::from(&mut Box::leak(node).data) }
+        BoxedNode { ptr: Unique::from(Box::leak(node)).cast() }
```

Comment:
https://github.com/rust-lang/rust/pull/78597 landed, so `ptr::raw_mut!` should work now.

Reply:
But I can't confirm that until #77611 lands (or you backport #78597 into miri's raw-retag branch), right?

Reply:
This is a rustc-only change that does not require a change on the Miri side. (Also miri's raw-retag branch landed.)

So you should be able to confirm this by using the Miri from today's nightly, and your extract btree test case (the latter is still needed to avoid https://github.com/rust-lang/rust/pull/77611).

Reply:
Oh right, I didn't realize these extracted btree test cases are truly stand-alone.

But I don't see any improvement. With Miri's nightly and the latest master, even the [miri #1608](https://github.com/rust-lang/miri/issues/1608) example still fails the same way.

Reply:
I guess you mean today's upcoming nightly, not the 2020-11-02 out there at the moment.

Reply:
The latest one out there at the moment is 2020-11-03. I meant 2020-11-04 which for some reason does not exist... (this is the rustup toolchain version, which is always one day ahead of what `rustc` reports -- rustc reports the day the merge commit was created, rustup reports the day the nightly was created)